lib/connection/connection_config.js

```diff
@@ -260,17 +221,29 @@ function ConnectionConfig(options, validateCredentials, qaMode, clientInfo) {
         protocol: proxyProtocol,
         noProxy: noProxy
       };
-    } else {
-      proxy =
-      {
-        host: proxyHost,
-        port: proxyPort,
-        protocol: proxyProtocol,
-        noProxy: noProxy
-      };
+    } 
+  
+    if (!proxy && GlobalConfig.isEnvProxyActive()) {
+      let proxyEnv = Util.getEnvVar('http_proxy') || Util.getEnvVar('https_proxy');
```

Comment:
in theory, http_proxy and https_proxy can be different (not likely, though). so we might need to think about handling it separately. `httpsAgent` will want HTTPS_PROXY, i guess. 
maybe we want to favor https_proxy over http_proxy; and put that to the left? in case both are defined, and are different